mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -512,6 +513,92 @@ object Vectors {
     squaredDistance
   }
 
+  private def dot(a : DenseVector, b : DenseVector) : Double = {
+    (a.toArray zip b.toArray).map(x => (x._1 * x._2)).sum
+  }
+
+  private def dot(a : SparseVector, b : DenseVector) : Double = {
+    (a.indices zip a.values).map(x => { b(x._1)*x._2 }).sum
```

Comment:
Fair enough, changed to breeze
I was using hadamard for Kij * (yi*yj) in an SVM implementation and was more convenient to use vector element wise when storing as a row matrix